utils/build-script-impl

```diff
@@ -565,21 +574,32 @@ function set_deployment_target_based_options() {
                 -DSWIFT_DARWIN_DEPLOYMENT_VERSION_WATCHOS="${DARWIN_DEPLOYMENT_VERSION_WATCHOS}"
                 -DLLVM_ENABLE_LIBCXX:BOOL=TRUE
             )
-
-            if [[ "${llvm_target_arch}" ]] ; then
-                llvm_cmake_options=(
-                    "${llvm_cmake_options[@]}"
-                    -DLLVM_TARGET_ARCH="${llvm_target_arch}"
-                )
-            fi
-
             ;;
         *)
-            echo "Unknown compiler deployment target: ${deployment_target}"
+            echo "Unknown host tools target: ${host}"
             exit 1
             ;;
     esac
 
+    if [[ "${llvm_target_arch}" ]] ; then
+        llvm_cmake_options+=(
+            -DLLVM_TARGET_ARCH:STRING="${llvm_target_arch}"
+        )
+    fi
+    if [[ "${llvm_host_triple}" ]] ; then
+        llvm_cmake_options+=(
+            -DLLVM_HOST_TRIPLE:STRING="${llvm_host_triple}"
+        )
+    fi
+    if [[ "${SWIFT_HOST_VARIANT}" ]] ; then
```

Comment:
When is `SWIFT_HOST_VARIANT` not set? In the switch statement above its set for all cases (all platforms).